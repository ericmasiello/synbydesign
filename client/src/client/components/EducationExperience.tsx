import * as React from 'react';
import styled from 'styled-components';
import Muted from './Muted';
import TypeSmall from './TypeSmall';
import TypeBase from './TypeBase';
import { HEADER_WEIGHTS } from '../styles/vars';

export interface Props extends Education {
  className?: string;
  tag?: Tag;
}

export const EducationExperience: React.SFC<Props> = props => {
  const {
    tag: Tag,
    className,
    institution,
    location,
    year,
    degree,
    ...rest
  } = props as Props;
  return (
    // @ts-ignore
    <Tag className={className} {...rest}>
      <TypeBase tag="h1" className="education__title">
        {institution}
        <div className="education__meta" data-meta>
          {location && <>{`${location} `}</>}
          {year && (
            <TypeSmall tag={Muted} data-year>
              ({year})
            </TypeSmall>
          )}
        </div>
      </TypeBase>
      {degree && <p data-degree>{degree}</p>}
    </Tag>
  );
};

EducationExperience.defaultProps = {
  tag: 'section',
};

export default styled(EducationExperience)`
  .education__title {
    font-weight: ${HEADER_WEIGHTS.bold};
  }

  .education__meta {
    font-weight: ${HEADER_WEIGHTS.medium};
  }
`;
