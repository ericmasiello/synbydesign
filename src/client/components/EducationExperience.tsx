import * as React from 'react';
import styled from 'styled-components';
import Tag from './Tag';

interface Props extends Education {
  className?: string;
  tag?: Tag;
}

interface DefaultProps {}

export const EducationExperience: React.SFC<Props> = props => {
  const {
    className,
    institution,
    location,
    year,
    degree,
    ...rest
  } = props as Props & DefaultProps;
  return (
    <Tag className={className} {...rest}>
      <h1>{institution}</h1>
      <div>
        {location && <p>{location}</p>}
        {year && <span>{year}</span>}
      </div>
      {degree && <p>{degree}</p>}
    </Tag>
  );
};

export default styled(EducationExperience)``;
