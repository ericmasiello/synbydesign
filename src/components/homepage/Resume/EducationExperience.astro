---
import classNames from 'classnames';
import Text from '../../Text.astro';
import type { Education } from './Resume';
import Spacer from './Spacer.astro';
import type { Props as SpacerProps } from './Spacer.astro';

export interface Props extends Omit<SpacerProps, 'slot'>, Education {}

const { as = 'article', degree, institution, location, year, className, ...rest } = Astro.props;
const classes = classNames('education-experience', className);
---

<Spacer class={classes} {...rest} as={as}>
  <Text as="h4" class="title">
    {institution}
    <div class="meta" data-meta>
      {location && <>{`${location} `}</>}
      {
        year && (
          <Text small muted data-year>
            ({year})
          </Text>
        )
      }
    </div>
  </Text>
  {degree && <p data-degree>{degree}</p>}
</Spacer>

<style>
  .education-experience {
    break-inside: avoid;
  }

  .title {
    font-weight: var(--font-body-weight-bold);
    margin-block-end: 0.25rem;
  }

  .meta {
    font-weight: var(--font-body-weight-medium);
  }
</style>
