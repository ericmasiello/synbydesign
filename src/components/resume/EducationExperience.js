import React from 'react';
import PropTypes from 'prop-types';
import { Text, Muted } from '../Type';
import * as styles from './EducationExperience.module.css';

function EducationExperience(props) {
  const { as: Component, className, institution, location, year, degree, ...rest } = props;
  return (
    <Component className={className} {...rest}>
      <Text component="h4" className={styles.title}>
        {institution}
        <div className={styles.meta} data-meta>
          {location && <React.Fragment>{`${location} `}</React.Fragment>}
          {year && (
            <Text small component={Muted} data-year>
              ({year})
            </Text>
          )}
        </div>
      </Text>
      {degree && <p data-degree>{degree}</p>}
    </Component>
  );
}

EducationExperience.defaultProps = {
  as: 'article',
};

EducationExperience.propTypes = {
  as: PropTypes.elementType,
};

export default EducationExperience;
