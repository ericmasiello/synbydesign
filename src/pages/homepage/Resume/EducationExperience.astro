---
import Text from '../../../components/Text.astro';
import type { Education } from './Resume';

export interface Props extends Omit<astroHTML.JSX.BaseHTMLAttributes, 'slot'>, Education {
  as?: keyof HTMLElementTagNameMap;
}

const { as: Element = 'article', degree, institution, location, year, ...rest } = Astro.props;
---

<Element {...rest}>
  <Text as="h4" class="title">
    {institution}
    <div class="meta" data-meta>
      {location && <>{`${location} `}</>}
      {
        year && (
          <Text small muted data-year>
            ({year})
          </Text>
        )
      }
    </div>
  </Text>
  {degree && <p data-degree>{degree}</p>}
</Element>

<style>
  .title {
    font-weight: var(--font-body-weight-bold);
  }

  .meta {
    font-weight: var(--font-body-weight-medium);
  }
</style>
