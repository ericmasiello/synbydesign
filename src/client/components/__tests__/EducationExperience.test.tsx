import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { EducationExperience } from '../EducationExperience';

const edu: Education = {
  institution: 'School Name',
};

it('should render', () => {
  const component = renderer.create(<EducationExperience {...edu} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should display additional classes', () => {
  const wrapper = shallow(
    <EducationExperience {...edu} className="my-class" />,
  );

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render location', () => {
  const wrapper = shallow(
    <EducationExperience {...edu} location="The location" />,
  );

  expect(
    wrapper
      .find('div')
      .text()
      .trim(),
  ).toBe('The location');
});

it('should render the year', () => {
  const wrapper = shallow(<EducationExperience {...edu} year="2018" />);

  expect(
    wrapper
      .find('div')
      .text()
      .trim(),
  ).toBe('2018');
});

it('should render degree', () => {
  const wrapper = shallow(
    <EducationExperience {...edu} degree="Master of None" />,
  );

  expect(
    wrapper
      .find('p')
      .text()
      .trim(),
  ).toBe('Master of None');
});
