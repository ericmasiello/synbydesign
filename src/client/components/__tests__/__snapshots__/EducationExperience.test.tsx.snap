// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<span
  className={undefined}
>
  <h1>
    School Name
  </h1>
  <div />
</span>
`;
